src/Nethermind/Nethermind.HealthChecks/FreeDiskSpaceChecker.cs

```diff
@@ -80,5 +65,44 @@ public async ValueTask DisposeAsync()
             await StopAsync(default);
             _timer.Dispose();
         }
+
+        public void EnsureEnoughFreeSpaceOnStart(ITimerFactory timerFactory)
+        {
+            float minAvailableSpaceThreshold = 2 * _healthChecksConfig.LowStorageSpaceShutdownThreshold;
+            if (_healthChecksConfig.LowStorageSpaceWarningThreshold > 0 && _healthChecksConfig.LowStorageSpaceWarningThreshold > _healthChecksConfig.LowStorageSpaceShutdownThreshold)
+                minAvailableSpaceThreshold = _healthChecksConfig.LowStorageSpaceWarningThreshold / 2.0f;
+
+            if (!IsEnoughDiskSpace(minAvailableSpaceThreshold))
+            {
+                ManualResetEventSlim mre = new(false);
+                using ITimer timer = timerFactory.CreateTimer(TimeSpan.FromMinutes(CheckPeriodMinutes));
+                timer.Elapsed += (t, e) =>
+                {
+                    if (IsEnoughDiskSpace(minAvailableSpaceThreshold))
+                        mre.Set();
+                };
+
+                timer.Start();
+                mre.Wait();
+            }
+        }
+
+        private bool IsEnoughDiskSpace(float minAvailableSpaceThreshold)
+        {
+            bool enoughSpace = true;
+            for (int index = 0; index < _drives.Length; index++)
```

Comment:
why does it check every single drive available?

Reply:
`_drives` is an already filtered collection based on the the DbPath. Generally takes care of any symbolic links, but in vast majority of cases, this will be a single drive.

Reply:
@tkstanczak to support multi-drive setups, when you have subfolders linked to different drives.